xarray/backends/netCDF4_.py

```diff
@@ -182,7 +182,10 @@ def _extract_nc4_variable_encoding(variable, raise_on_invalid=False,
 def _open_netcdf4_group(filename, mode, group=None, **kwargs):
     import netCDF4 as nc4
 
-    ds = nc4.Dataset(filename, mode=mode, **kwargs)
+    try:
+        ds = nc4.Dataset(filename, mode=mode, **kwargs)
+    except OSError as e:
+        raise OSError("Error opening %r" % filename, e)
```

Comment:
I don't think this is using the `OSError` constructor properly -- the first argument is supposed to be an error number:
https://docs.python.org/2/library/exceptions.html#exceptions.EnvironmentError

Reply:
You're right. I wasn't aware of this. I'm wondering if we should just raise an IOError here?

Reply:
That does feel like a better error type for most cases, but I'm not sure about every case. Presumably there's some reason why netCDF4 raises OSError...

Reply:
Thanks, I'll play around with this a bit more.